explorer/ast/declaration.cpp

```diff
@@ -15,104 +15,112 @@ using llvm::cast;
 Declaration::~Declaration() = default;
 
 void Declaration::Print(llvm::raw_ostream& out) const {
+  PrintDepth(-1, 0, out);
+}
+void Declaration::PrintDepth(int depth, int indent_num_spaces,
+                             llvm::raw_ostream& out) const {
+  if (depth == 0) {
+    out.indent(indent_num_spaces) << " ... ";
+    return;
+  }
+
   switch (kind()) {
     case DeclarationKind::NamespaceDeclaration:
-      PrintID(out);
-      out << ";";
+      out.indent(indent_num_spaces) << PrintAsID(*this) << ";";
       break;
     case DeclarationKind::InterfaceDeclaration:
     case DeclarationKind::ConstraintDeclaration: {
       const auto& iface_decl = cast<ConstraintTypeDeclaration>(*this);
-      PrintID(out);
-      out << " {\n";
+      out.indent(indent_num_spaces) << PrintAsID(*this) << " {\n";
       for (Nonnull<Declaration*> m : iface_decl.members()) {
-        out << *m;
+        out.indent(indent_num_spaces + 2) << *m << "\n";
       }
-      out << "}\n";
+      out.indent(indent_num_spaces) << "}";
       break;
     }
     case DeclarationKind::ImplDeclaration: {
       const auto& impl_decl = cast<ImplDeclaration>(*this);
-      PrintID(out);
-      out << " {\n";
+      out.indent(indent_num_spaces) << PrintAsID(impl_decl) << " {\n";
       for (Nonnull<Declaration*> m : impl_decl.members()) {
-        out << *m;
+        m->PrintDepth(-1, indent_num_spaces + 2, out);
```

Comment:
Shouldn't cases like this be passing `depth - 1` if you're adding `depth` as a parameter?

Reply:
I wasn't exactly sure how the `depth` thing was working by the time I wrote this code.
I'm waiting for your response in the **#explorer** on discord regarding if we need to keep `PrintDepth` or not before working on this change.

Reply:
Let's go ahead with that, maybe also renaming PrintDepth to PrintIndented or similar?